lampo-bdk-wallet/src/lib.rs

```diff
@@ -73,32 +80,44 @@ impl BDKWalletManager {
         xprv: PrivateKey,
         channel_keys: Option<String>,
     ) -> error::Result<(Wallet, LampoKeys)> {
+    // ) -> error::Result<(PersistedWallet<Store<ChangeSet>>, LampoKeys)> {
+
         let ldk_keys = if channel_keys.is_some() {
             LampoKeys::with_channel_keys(xprv.inner.secret_bytes(), channel_keys.unwrap())
         } else {
             LampoKeys::new(xprv.inner.secret_bytes())
         };
 
         // FIXME: Get a tmp path
-        let db = Store::open_or_create_new("lampo".as_bytes(), "/tmp/onchain")?;
+        // let db: Store<ChangeSet> = Store::open_or_create_new("lampo".as_bytes(), "/tmp/onchain")?;
         let network = match xprv.network.to_string().as_str() {
-            "bitcoin" => bdk::bitcoin::Network::Bitcoin,
-            "testnet" => bdk::bitcoin::Network::Testnet,
-            "signet" => bdk::bitcoin::Network::Signet,
-            "regtest" => bdk::bitcoin::Network::Regtest,
+            "bitcoin" => bdk_wallet::bitcoin::Network::Bitcoin,
+            "testnet" => bdk_wallet::bitcoin::Network::Testnet,
+            "signet" => bdk_wallet::bitcoin::Network::Signet,
+            "regtest" => bdk_wallet::bitcoin::Network::Regtest,
             _ => unreachable!(),
         };
         let key = Xpriv::new_master(network, &xprv.inner.secret_bytes())?;
-        let key = ExtendedKey::from(key);
-        let wallet = Wallet::new(Bip84(key, KeychainKind::External), None, db, network)?;
+        let key: ExtendedKey<Legacy> = ExtendedKey::from(key);
+        let key = key
+            .into_xprv(network)
+            .ok_or(error::anyhow!("wrong convertion to a private key"))?;
+        // let wallet = Wallet::create(Bip84(key, KeychainKind::External), None, db, network);
+        let wallet = Wallet::create(
+            Bip84(key.clone(), KeychainKind::External),
+            Bip84(key, KeychainKind::Internal),
+        )
+            .network(network)
+            // .create_wallet(&mut db)?;
+            .create_wallet_no_persist()?;
```

Comment:
Why are you removing the persistence of the wallet?

Reply:
this is a simplification step, to pass compiler and move forward (I put persistence is a TODO)
since bdk_wallet has `PersistedWallet` special type opposed to just `Wallet`, need to reflect those typings and do proper sync implementation